### ISSUE TITLE
eager load of links

### ISSUE THREAD

**Issue by DJafari** [isar/isar#369]
is possible support eager load links for better performance ( solve n + 1 query ) ?

in mysql eager load works something like this :

```dart
@Collection()
class Teacher {
  int? id;

  late String name;
}

@Collection()
class Student {
  int? id;

  late String name;

  final teachers = IsarLink<Teacher>();
}
```

```
SELECT * from `teachers` // [ { id: 3, name: Test1 }, { id: 5, name: Test2 } ]
```

```
SELECT * FROM `students` WHERE `teacher_id` IN (3,5)
```

**Comment by DJafari** [comment]
imagine i have a listview and i want show all students with teacher`s name, how can doing that ?

 i must load each teacher of student and shows them ?
imagine we have 1000 student with one teacher, it must run 1001 query

eagerload must be optional, it's mean only when we want using something like above, i must use it

something like this :

```
isar.students.with(['teachers']).find();
```

when using `with` isar must autoload links that we provided in parameter

**Comment by leisim** [comment]
Fixed in `3.0.0-dev.8`
